wordpress/WordpressClient.ts

```diff
@@ -0,0 +1,113 @@
+// Copyright (c) 2022. Heusala Group Oy <info@heusalagroup.fi>. All rights reserved.
+
+import { LogLevel } from "../types/LogLevel";
+import { LogService } from "../LogService";
+import { HttpService } from "../HttpService";
+import { isWordpressPagesDTO, WordpressPageListDTO } from "./dto/WordpressPageListDTO";
+import { isWordpressReferencesDTO, WordpressReferenceListDTO } from "./dto/WordpressReferenceListDTO";
+import { isWordpressUserProfilesDTO, WordpressUserProfileListDTO } from "./dto/WordpressUserProfileListDTO";
+
+const LOG = LogService.createLogger('WordpressClient');
+
+export interface WordpressContentInterface {
+    readonly pages?: Promise<WordpressPageListDTO>;
+    readonly references?: Promise<WordpressReferenceListDTO>;
+    readonly profiles?: Promise<WordpressUserProfileListDTO>;
+}
+
+export class WordpressClient {
+
+    public static setLogLevel(level: LogLevel) {
+        LOG.setLogLevel(level);
+        HttpService.setLogLevel(level);
+    }
+
+    private static _defaultUrl: string = '/';
+    private static _endpoint: string = '/';
+
+    private readonly _url: string;
+    private readonly _endpoint: string;
```

Comment:
Is there reason that this is configurable through constructor and not constant?

Anyway, either change it (one of these):

1. Change to use constant if these end points are static
2. Multiple values for endpoints if these endpoints may be dynamic (different for each WP installation / use case and there isn't better way to configure it)
3. Multiple classes for each method/use case using different end point variable if it's dynamic

Code shouldn't have methods that aren't usable. Right now programmer would need to initiate three instances of this class and know which method works and which does not.

Reply:
### Changed dynamic endpoints to constants.

https://github.com/heusalagroup/fi.hg.core/pull/13/commits/683c351ff3088b64e0a25dc7f8e35bd6eb6492a6#r990944031